ignite/base/events_driven.py

```diff
@@ -0,0 +1,267 @@
+import functools
+import logging
+import weakref
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Mapping, Optional, Union
+
+from ignite.engine.events import CallableEventWithFilter, EventEnum, Events, EventsList, RemovableEventHandle
+from ignite.engine.utils import _check_signature
+
+
+class EventsDriven:
+    """Base class for events-driven engines without state.
+
+    """
```

Comment:
If we make this class public and expose it in the docs. We need to make this docstring better and provide examples

Reply:
I agree with that, thanks!